src/utils/transactions.js

```diff
@@ -40,7 +51,77 @@ function getTxPendingTime (coordinatorState, isL1, timestamp) {
   return timeLeftToForgeInMinutes > 0 ? timeLeftToForgeInMinutes : 0
 }
 
+/**
+   * Checks whether an amount is supported by the compression
+   * used in the Hermez network
+   * @param {Number} amount - Selector amount
+   * @returns {Boolean} Whether it is valid
+   */
+function isTransactionAmountCompressedValid (amount) {
+  try {
+    const compressedAmount = HermezCompressedAmount.compressAmount(amount)
+    const decompressedAmount = HermezCompressedAmount.decompressAmount(compressedAmount)
+
+    return amount.toString() === decompressedAmount.toString()
+  } catch (e) {
+    return false
+  }
+}
+
+/**
+ * Fixes the transaction amount to be sure that it would be supported by Hermez
+ * @param {BigNumber} amount - Transaction amount to be fixed
+ * @returns fixedTxAmount
+ */
+function fixTransactionAmount (amount) {
+  const fixedTxAmount = HermezCompressedAmount.decompressAmount(
+    HermezCompressedAmount.floorCompressAmount(amount)
+  )
+
+  return BigNumber.from(fixedTxAmount)
+}
+
+/**
+ * Calculates the max amoumt that can be sent in a transaction
+ * @param {TxType} txType - Transaction type
+ * @param {BigNumber} maxAmount - Max amount that can be sent in a transaction (usually it's an account balance)
+ * @param {Object} token - Token object
+ * @param {Number} l2Fee - Transaction fee
+ * @param {BigNumber} gasPrice - Ethereum gas price
+ * @returns maxTxAmount
+ */
+function getMaxTxAmount (txType, maxAmount, token, l2Fee, gasPrice) {
+  const maxTxAmount = (() => {
+    switch (txType) {
+      case TxType.ForceExit: {
+        return maxAmount
+      }
+      case TxType.Deposit: {
+        const depositFee = getDepositFee(token, gasPrice)
+        const newMaxAmount = maxAmount.sub(depositFee)
+
+        return newMaxAmount.gt(0)
+          ? newMaxAmount
+          : BigNumber.from(0)
+      }
+      default: {
+        const l2FeeBigInt = parseUnits(l2Fee.toFixed(token.decimals), token.decimals)
+        const newMaxAmount = BigNumber.from(getMaxAmountFromMinimumFee(l2FeeBigInt, maxAmount).toString())
+
+        return newMaxAmount.gt(0)
```

Comment:
I think due to the nature of `getMaxAmountFromMinimumFee`, this value will never be < 0